app/controllers/charges_controller.rb

```diff
@@ -11,6 +11,11 @@ def create
     return unless verify_recaptcha
     return if @amount.nil? || @amount.zero? || @amount.negative?
 
+    if @amount < 500
```

Comment:
I think a better approach is to use [ActiveModel:Errors](https://api.rubyonrails.org/classes/ActiveModel/Errors.html) in the model.